spinalcordtoolbox/deepseg/models.py

```diff
@@ -12,7 +12,7 @@
 
 import spinalcordtoolbox as sct
 from spinalcordtoolbox import download
-from spinalcordtoolbox.utils import stylize
+from spinalcordtoolbox.utils import stylize, __deepseg_dir__
```

Comment:
FYI, this is a strange in-between.

The only direct content of the `spinalcordtoolbox` module is a definition of `__deepseg_dir__` (and a few other strings), via an import statement, so that seems like it clearly conveys intent; in other words, `sct.__deepseg_dir__` doesn't look accidental.

Also, `spinalcordtoolbox.utils` similarly imports `__deepseg_dir__` from `spinalcordtoolbox.utils.sys`. Again, probably not accidental.

So, I'm not against this particular change, just not certain what it buys us. Maybe this: if we end up consistently importing directly from the actual modules where things are defined, then we can get rid of the `# noqa` imports in `__init__.py` and `utils/__init__.py` entirely.

Reply:
> The only direct content of the `spinalcordtoolbox` module is a definition of `__deepseg_dir__` (and a few other strings), via an import statement, so that seems like it clearly conveys intent; in other words, `sct.__deepseg_dir__` doesn't look accidental.
> 
> Also, `spinalcordtoolbox.utils` similarly imports `__deepseg_dir__` from `spinalcordtoolbox.utils.sys`. Again, probably not accidental.

This is a very good point! With the way `spinalcordtoolbox/__init__.py` and `spinalcordtoolbox/utils/__init__.py` are written, accessing `__deepseg_dir__` through `sct.__deepseg_dir__` does seem to be intended usage. :thinking: 

Admittedly, I think I blindly went after every instance of `import spinalcordtoolbox as sct` without thinking hard enough about the situations where the import is risky.

Reply:
> then we can get rid of the `# noqa` imports in `__init__.py` and `utils/__init__.py` entirely.

To look at these linting warnings a little further, they are:

code | sample message
-- | --
F401 | module imported but unused
F403 | ‘from module import *’ used; unable to detect undefined names

That said, it looks like F401s in `__init__.py` files are [common enough to have SO questions for them](https://stackoverflow.com/questions/8427701/how-to-ignore-pyflakes-errors-imported-but-unused-in-the-init-py-file) (though I don't know if I can say the same for F403 star-importing). So, I'm not sure if that alone is a good enough justification for clearing out the imports in `__init__.py`.

Reply:
All considered, I think it might be best to revert my change to `__deepseg_dir__`, then save further discussion about the `__init__.py` files (and how we expose various functions/modules/prorties) for a follow-up issue?